code/game/objects/items/storage/holsters.dm

```diff
@@ -220,6 +220,124 @@
+		/obj/item/mortar_kit,
+	)
+
+	sprite_sheets = list(
+		"Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
+		"Sterling Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
+		"Chilvaris Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
+		"Hammerhead Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
+		"Ratcher Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
+		)
+
+
+/obj/item/storage/holster/backholster/mortar/full/Initialize()
+	. = ..()
+	var/obj/item/new_item = new /obj/item/mortar_kit(src)
+	INVOKE_ASYNC(src, PROC_REF(handle_item_insertion), new_item)
+
+/obj/item/storage/holster/backholster/flamer
+	name = "\improper TGMC flamethrower bag"
+	desc = "This backpack can carry its accompanying flamethrower as well as a modest general storage capacity. Automatically refuels it's accompanying flamethrower."
+	icon_state = "pyro_bag"
+	item_state = "pyro_bag"
+	w_class = WEIGHT_CLASS_BULKY
+	storage_slots = null
+	max_storage_space = 16
+	max_w_class = WEIGHT_CLASS_NORMAL
+	access_delay = 0
+	holsterable_allowed = list(/obj/item/weapon/gun/flamer/big_flamer/marinestandard/engineer,)
+	bypass_w_limit = list(/obj/item/weapon/gun/flamer/big_flamer/marinestandard/engineer,)
+	storage_type_limits = list(/obj/item/weapon/gun/flamer/big_flamer/marinestandard/engineer = 1,)
+		///The internal fuel tank
+	var/obj/item/ammo_magazine/flamer_tank/internal/tank
+
+	///The linked flamerthrower that cannot be dropped
+
+	sprite_sheets = list(
+		"Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
+		"Sterling Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
+		"Chilvaris Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
+		"Hammerhead Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
+		"Ratcher Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
+		)
+
+/obj/item/storage/holster/backholster/flamer/Initialize()
+	. = ..()
+	tank = new
+	update_icon()
+
+/obj/item/storage/holster/backholster/flamer/MouseDrop_T(obj/item/W, mob/living/user)
+	. = ..()
+	if(istype(W,/obj/item/ammo_magazine/flamer_tank))
+		refuel(W, user)
+
+/obj/item/storage/holster/backholster/flamer/afterattack(obj/O as obj, mob/user as mob, proximity)
+	. = ..()
+	//uses the tank's proc to refuel
+	if(istype(O, /obj/structure/reagent_dispensers/fueltank))
+		tank.afterattack(O, user)
+	if(istype(O,/obj/item/ammo_magazine/flamer_tank))
+		refuel(O, user)
+
+/obj/item/storage/holster/backholster/flamer/handle_item_insertion(obj/item/item, prevent_warning = 0, mob/user)
+	. = ..()
+	if (holstered_item == item)
+		var/obj/item/weapon/gun/flamer/big_flamer/marinestandard/engineer/flamer = item
+		refuel(flamer.chamber_items[1], user)
+		flamer.update_ammo_count()
+
+/* Used to refuel the attached FL-86 flamer when it is put into the backpack
+ *
+ * param1 - The flamer tank, the actual tank we are refilling
+ * param2 - The person wearing the backpack
+*/
+/obj/item/storage/holster/backholster/flamer/proc/refuel(obj/item/W, mob/living/user)
```

Comment:
this procs relies on W being a flametank, so the arg should be typecaste as such, rather than in the proc. It should also istype to actually ensure something else hasn't been passed.